test/e2e/aggregated_clusteroperator_test.go

```diff
@@ -0,0 +1,145 @@
+				err := c.Get(ctx, types.NamespacedName{Name: "clusteroperators.config.openshift.io"}, &crd)
+				if err != nil {
+					return false, err
+				}
+				return crdReady(&crd.Status), nil
+			}).Should(BeTrue())
+		})
+
+		AfterEach(func() {
+			Expect(c.Delete(ctx, ns)).To(BeNil())
+			Eventually(func() error {
+				err := c.Delete(ctx, &crd)
+				if apierrors.IsNotFound(err) {
+					return nil
+				}
+				return err
+			}).Should(Succeed())
+		})
+
+		When("installing a series of POs successfully", func() {
+			var (
+				po *platformv1alpha1.PlatformOperator
+			)
+			BeforeEach(func() {
+				po = &platformv1alpha1.PlatformOperator{
+					ObjectMeta: metav1.ObjectMeta{
+						GenerateName: "prometheus-operator",
+					},
+					Spec: platformv1alpha1.PlatformOperatorSpec{
+						Package: platformv1alpha1.Package{
+							Name: "prometheus-operator",
+						},
+					},
+				}
+				Expect(c.Create(ctx, po)).To(BeNil())
+			})
+			AfterEach(func() {
+				Expect(c.Delete(ctx, po)).To(BeNil())
+			})
+
+			It("should result in a successful application", func() {
+				Eventually(func() (*metav1.Condition, error) {
+					if err := c.Get(ctx, client.ObjectKeyFromObject(po), po); err != nil {
+						return nil, err
+					}
+					return meta.FindStatusCondition(po.Status.Conditions, platformtypes.TypeApplied), nil
+				}).Should(And(
+					Not(BeNil()),
+					WithTransform(func(c *metav1.Condition) string { return c.Type }, Equal(platformtypes.TypeApplied)),
+					WithTransform(func(c *metav1.Condition) metav1.ConditionStatus { return c.Status }, Equal(metav1.ConditionTrue)),
+					WithTransform(func(c *metav1.Condition) string { return c.Reason }, Equal(platformtypes.ReasonApplySuccessful)),
+					WithTransform(func(c *metav1.Condition) string { return c.Message }, ContainSubstring("Successfully applied the desired olm.bundle content")),
+				))
+			})
+
+			It("should eventually report a healthy CO status back to the CVO", func() {
+				Eventually(func() (*configv1.ClusterOperatorStatusCondition, error) {
+					if err := c.Get(ctx, client.ObjectKeyFromObject(&co), &co); err != nil {
+						return nil, err
+					}
+					for _, cond := range co.Status.Conditions {
+						if cond.Type == configv1.OperatorAvailable {
+							return &cond, nil
+						}
+						continue
+					}
+					return nil, nil
+				}).Should(And(
+					Not(BeNil()),
+					WithTransform(func(c *configv1.ClusterOperatorStatusCondition) configv1.ClusterStatusConditionType { return c.Type }, Equal(configv1.OperatorAvailable)),
+					WithTransform(func(c *configv1.ClusterOperatorStatusCondition) configv1.ConditionStatus { return c.Status }, Equal(configv1.ConditionTrue)),
+					WithTransform(func(c *configv1.ClusterOperatorStatusCondition) string { return c.Reason }, Equal("POsHealthy")),
+					WithTransform(func(c *configv1.ClusterOperatorStatusCondition) string { return c.Message }, ContainSubstring("All POs in a successful state")),
+				))
+
+			})
+		})
```

Comment:
Can we add two test cases here?

1. One PO that fails reports an unhealthy CO to the CVO
2. One PO fails out of multiple succeeding POs which reports an unhealthy CO to CVO

Reply:
I think we'd want to initially test out the following scenarios:

- Happy path
- No POs in the cluster
- A PO that's failing to find a bundle from the desired package name
- A mixture of successful and failed POs results in an unavailable condition

These are likely closer to unit tests, but we can work towards refactoring those checks further down the line.

Reply:
Done. Added all of those test cases in the latest force push.